nmap_dns.cc

```diff
@@ -784,7 +793,10 @@ static void read_evt_handler(nsock_pool nsp, nsock_event evt, void *) {
               }
               output_summary();
               stat_ok++;
+              /*Since we have got the answer we can avoid iterating over all answers*/
```

Comment:
@tremblerz With the current code, we cannot associate the later PTR record with our request because the name will be different than what we sent. That is the purpose of the CNAME record. It is as if we asked, "Who lives in the blue house?" and were told, "The blue house is 123 Main Street. John lives at 123 Main Street." Without understanding that "123 Main Street" refers to the same location as "the blue house," we can't know that "John" is the answer to our question. So for now, as soon as we see a CNAME, we can hand it off to system resolver. A later change we discussed would be chaining those records together to get the complete answer.